NeoML/src/TraditionalML/SparseFloatVector.cpp

```diff
@@ -44,12 +44,23 @@ CSparseFloatVector::CSparseFloatVectorBody::CSparseFloatVectorBody( int bufferSi
 CSparseFloatVector::CSparseFloatVectorBody::CSparseFloatVectorBody( const CSparseFloatVectorDesc& desc ) :
 	BufferSize( desc.Size )
 {
-	Desc.Size = desc.Size;
-	IndexesBuf.SetSize( BufferSize );
-	ValuesBuf.SetSize( BufferSize );
-	::memcpy( IndexesBuf.GetPtr(), desc.Indexes, Desc.Size * sizeof( int ) );
-	::memcpy( ValuesBuf.GetPtr(), desc.Values, Desc.Size * sizeof( float ) );
-
+	if( desc.Indexes == nullptr ) {
+		IndexesBuf.SetBufferSize( BufferSize );
```

Comment:
It looks like this may consume a lot of memory if desc contained lots of zeros (the buffer size will be way bigger than needed).

Reply:
ok.. here we have 2 options:
1. count element before allocating buffer
2. don't reserve space and extend buffer by CArray strategy.

think 1 is better.

Reply:
done

Reply:
Agree